lib/IAMClient.js

```diff
@@ -55,16 +54,14 @@ class VaultClient {
         this._ca = ca;
         this.useHttps = (useHttps === true);
         if (this.useHttps) {
-            this._agent = new HttpsAgent({
+            this._agent = new HttpsAgent.Agent({
                 ca: ca ? [ca] : undefined,
-                keepAlive: true,
                 requestCert: true,
                 rejectUnauthorized: !(ignoreCa === true),
                 freeSocketTimeout: constants.httpClientFreeSocketTimeout,
```

Comment:
I don't think the new `httpagent` module supports the `freeSocketTimeout` option, and it would be important to check whether the new behavior is acceptable without this option supported by `agentkeepalive` (which was introduced to fix https://scality.atlassian.net/browse/S3C-3114).

If this change is neeeded for performance, it could be necessary to merge the `freeSocketTimeout` option support to `httpagent` or find an alternative way to avoid the ECONNRESET issue.

Reply:
Thanks @jonathan-gramain that's true and even if I did not faced this problem yet, we want to be safe. To both keep this idea of enforcing the `maxSockets` option and the agentkeepalive without reinventing the wheel, I did a PR in `httpagent` to export the default configuration: https://github.com/scality/httpagent/pull/6

Meanwhile, I rebased this PR to keep `agentkeepalive` but only add the default configuration on top of it: https://github.com/scality/vaultclient/pull/420/files#diff-86a8509a2ad22b19f93123bda4efc46cfa71b4c2bbfded93e5f6e1a25e69f180R65

I also did it this way in a similar PR I have, in bucketclient, that, I believe, includes the same fix as agentkeepalive is used: https://github.com/scality/bucketclient/pull/248

I will bump the httpagent version before merging.

Reply:
Update: following our discussion, httpagent was updated to extend agentkeepalive by default. This way we enforce all the keep alive options by default along with maxSockets, and we can remove agentkeepalive from this project's dependencies.

I updated the PR accordingly, using the latest version of httpagent.